client/widgets/VideoWidget.cpp

```diff
@@ -91,6 +113,34 @@ void VideoWidgetBase::stopAudio()
 	}
 }
 
+std::string VideoWidgetBase::getSubTitleLine(double timestamp)
+{
+	if(srtContent.empty())
+		return "";
+	
+    std::regex exp("^\\s*(\\d+:\\d+:\\d+,\\d+)[^\\S\\n]+-->[^\\S\\n]+(\\d+:\\d+:\\d+,\\d+)((?:\\n(?!\\d+:\\d+:\\d+,\\d+\\b|\\n+\\d+$).*)*)", std::regex::multiline);
+    std::smatch res;
+
+    std::string::const_iterator searchStart(srtContent.cbegin());
+    while (std::regex_search(searchStart, srtContent.cend(), res, exp))
+    {
+		std::vector<std::string> timestamp1Str;
+		boost::split(timestamp1Str, static_cast<std::string>(res[1]), boost::is_any_of(":,"));
+		std::vector<std::string> timestamp2Str;
+		boost::split(timestamp2Str, static_cast<std::string>(res[2]), boost::is_any_of(":,"));
+		double timestamp1 = std::stoi(timestamp1Str[0]) * 3600 + std::stoi(timestamp1Str[1]) * 60 + std::stoi(timestamp1Str[2]) + (std::stoi(timestamp1Str[3]) / 1000.0);
+		double timestamp2 = std::stoi(timestamp2Str[0]) * 3600 + std::stoi(timestamp2Str[1]) * 60 + std::stoi(timestamp2Str[2]) + (std::stoi(timestamp2Str[3]) / 1000.0);
+        std::string text = res[3];
+		text.erase(0, text.find_first_not_of("\r\n\t "));
+
+		if(timestamp > timestamp1 && timestamp < timestamp2)
```

Comment:
would be much better to parse string content only once and then search your data structure

Reply:
oh, didn't notice your TODO